storm-core/src/jvm/org/apache/storm/trident/windowing/WindowTridentProcessor.java

```diff
@@ -163,11 +163,14 @@ public void finishBatch(ProcessorContext processorContext) {
         Iterable<Object> triggerValues = null;
 
         if (retriedAttempt(batchId)) {
-            pendingTriggerIds = (List<Integer>) windowStore.get(inprocessTriggerKey(batchTxnId));
-            for (Integer pendingTriggerId : pendingTriggerIds) {
-                triggerKeys.add(triggerKey(pendingTriggerId));
+            Object triggerIds = windowStore.get(inprocessTriggerKey(batchTxnId));
```

Comment:
@wangperry Can you remove creating one more variable. You need to wrap with a simple null check like below.

```java
if(pendingTriggerIds != null)  {
    for (Integer pendingTriggerId : pendingTriggerIds) {
      triggerKeys.add(triggerKey(pendingTriggerId));
   }
   triggerValues = windowStore.get(triggerKeys);
}
```